stix_shifter/stix_transmission/src/modules/guardium/guardiumapiclient.py

```diff
@@ -0,0 +1,448 @@
+        # Subroto -- Guardium does not have ping facility
+        # We test if we can get the access token if we can then success = true
+        #
+        logging.info("--------- Guardium ping_box ------\n")
+        respObj = Response()
+        if (self.fetch_accessToken()):
+            respObj.code = "200"
+            respObj.error_type = ""
+            respObj.status_code = 200
+            content = '{"status":"OK", "data": {"message": "Service is up."}}'
+            respObj._content = bytes(content, 'utf-8')
+        else:
+            respObj.code = "503"
+            respObj.error_type = "Service Unavailable"
+            respObj.status_code = 503
+            content = '{"status":"Failed", "data": {"message": "Service is down."}}'
+            respObj._content = bytes(content, 'utf-8')
+        # changed by Subroto: https://<server_ip>/api/ariel/searches/<search_id>
+        # endpoint = self.endpoint_start + "searches/" + search_id
+        # return self.client.call_api(endpoint, 'GET')
+        return ResponseWrapper(respObj)
+        # Sends a GET request
+        # to https://<server_ip>/api/help/resources
+        # endpoint = 'api/help/resources'  # no 'ariel' in the path
+        # return self.client.call_api(endpoint, 'GET')
+
+    def get_databases(self):
+        # Sends a GET request
+        # to  https://<server_ip>/api/ariel/databases
+        endpoint = self.endpoint_start + 'databases'
+        return self.client.call_api(endpoint, 'GET')
+
+    def get_database(self, database_name):
+        # Sends a GET request
+        # to https://<server_ip>/api/ariel/databases/<database_name>
+        endpoint = self.endpoint_start + 'databases' + '/' + database_name
+        return self.client.call_api(endpoint, 'GET')
+
+    def isTimestampValid(self, tstamp):
+        if tstamp is not None:
+            if(tstamp > (datetime.datetime.now()).timestamp()):
+                return True
+        return False
+#
+    def get_credential(self):
+        # Subroto -- Assumption: credential object will contain the full
+        # guardium credential for the call.  There is an option to read it from a file too
+        # proviee the path name if we want to read this credential from a file
+        if self.credential is None:
+            self.credential = self.read_credential_from_file()
+#
+        jCred = self.credential
+#
+        data = 'client_id=' + jCred.get("client_id") + '&grant_type=' + str(jCred.get("grant_type")) + \
+               '&client_secret=' + str(jCred.get("client_secret")) + '&username=' + str(jCred.get("username")) + \
+               '&password=' + str(jCred.get("password"))
+        print("credential data: " + data)
+        return data
+#
+    def read_credential_from_file(self):
+        # Subroto -- Assumption: credential from a file if the setup is done 
+        # self.credential_file is None.  Do not read from the file
+        username = self.connection.get("username", "admin")
```

Comment:
username would come from the `configuration`, not the connection. It would actually be `configuration.get('auth').get('username')`. Also, you shouldn't be defaulting to admin, if the configuration doesn't have a username, and that's required to access the data source, then the configuration is invalid and you should be throwing an error. All user-specific information is stored in the configuration.

Reply:
Will all credential parameters come from configuration.get('auth').get('username'), etc.?

Reply:
All credentials are stored in the auth object that is in the configuration object